ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -193,21 +191,44 @@ func (scbuilder *SiteConfigBuilder) getExtraManifest(dataMap map[string]interfac
 				dataMap[filename] = value
 			}
 		} else {
-			manifestFile, err := scbuilder.fHandler.ReadSourceFile(filePath)
+			manifestFile, err := ReadFile(filePath)
+			if err != nil {
+				return dataMap, err
+			}
+			manifestFileStr := string(manifestFile)
+			dataMap[file.Name()] = manifestFileStr
+		}
+	}
+
+	// Adding End User Extra-manifest
+	if clusterSpec.ExtraManifestPath != "" {
+		files, err = GetFiles(clusterSpec.ExtraManifestPath)
+		if err != nil {
+			return dataMap, err
+		}
+		for _, file := range files {
+			if file.IsDir() {
+				continue
+			}
+
+			filePath := clusterSpec.ExtraManifestPath + "/" + file.Name()
+			manifestFile, err := ReadFile(filePath)
 			if err != nil {
 				return dataMap, err
 			}
+
 			manifestFileStr := string(manifestFile)
 			dataMap[file.Name()] = manifestFileStr
```

Comment:
This will allow a user to completely override any of the files we provide if they put one with the same filename in their override directory.

Not saying this is a bad thing, but I want to make sure we introduce this facility deliberately, not accidentally.

Reply:
okay, I believe this will get better shape as we discuss it in future enhancement.

Reply:
That's a "feature" that would be really hard to take away. What if we prevent the user from overwriting our extra-manifest and then relax that as the future enhancement? Or do you see the need to allow overwriting?